src/invidious/yt_backend/youtube_api.cr

```diff
@@ -666,6 +670,51 @@ module YoutubeAPI
     return initial_data
   end
 
+  ####################################################################
+  # _post_invidious_companion(endpoint, data)
+  #
+  # Internal function that does the actual request to Invidious companion
+  # and handles errors.
+  #
+  # The requested data is an endpoint (URL without the domain part)
+  # and the data as a Hash object.
+  #
+  def _post_invidious_companion(
+    endpoint : String,
+    data : Hash
+  ) : Hash(String, JSON::Any)
+    headers = HTTP::Headers{
+      "Content-Type"  => "application/json; charset=UTF-8",
+      "Authorization" => "Bearer #{CONFIG.invidious_companion_key}",
+    }
+
+    # Logging
+    LOGGER.debug("Invidious companion: Using endpoint: \"#{endpoint}\"")
+    LOGGER.trace("Invidious companion: POST data: #{data}")
+
+    # Send the POST request
+
+    begin
+      invidious_companion = CONFIG.invidious_companion.sample
+      response = make_client(invidious_companion.private_url,
```

Comment:
Also if a HTTP proxy is configured in the Invidious configs then Invidious will also attempt to connect to the companion's private URL with the HTTP proxy

You'll need to add an argument in the `make_client` function to skip using the http proxy